src/NUnitFramework/framework/Internal/ExceptionHelper.cs

```diff
@@ -33,6 +34,41 @@ namespace NUnit.Framework.Internal
     /// </summary>
     public class ExceptionHelper
     {
+#if !NET_4_5 && !PORTABLE
+        private static readonly Action<Exception> PreserveStackTrace;
+
+        static ExceptionHelper()
+        {
+            var method = typeof(Exception).GetMethod("InternalPreserveStackTrace", BindingFlags.Instance | BindingFlags.NonPublic);
+
+            if (method != null)
+            {
+                try
+                {
+                    PreserveStackTrace = (Action<Exception>)Delegate.CreateDelegate(typeof(Action<Exception>), method);
+                    return;
+                }
+                catch (InvalidOperationException) { }
+            }
+            PreserveStackTrace = _ => { };
+        }
+#endif
+
+        /// <summary>
+        /// Rethrows an exception, preserving its stack trace
+        /// </summary>
+        /// <param name="exception">The exception to rethrow</param>
+        public static void Rethrow(Exception exception)
```

Comment:
What I missed in  reviewing this is that it broke the compact framework build. Oddly (?) nobody noticed until I just came to make another change. No need to worry about it particularly or even to build for CF yourself. Just bare in mind that NETCF is one of our builds when you are working with #if statements. I ended up removing this entire method and the following from NETCF.